doc/mgmt/OpenConfig_PortChannel_Interface.md

```diff
@@ -0,0 +1,279 @@
+# OpenConfig support for PortChannel (aggregate) interface.
+
+# High Level Design Document
+#### Rev 0.1
+
+# Table of Contents
+  * [List of Tables](#list-of-tables)
+  * [Revision](#revision)
+  * [About This Manual](#about-this-manual)
+  * [Scope](#scope)
+  * [Definition/Abbreviation](#definitionabbreviation)
+  * [1 Feature Overview](#1-feature-overview)
+    * [1.1 Requirements](#11-requirements)
+      * [1.1.1 Functional Requirements](#111-functional-requirements)
+      * [1.1.2 Configuration and Management Requirements](#112-configuration-and-management-requirements)
+    * [1.2 Design Overview](#12-design-overview)
+      * [1.2.1 Basic Approach](#121-basic-approach)
+      * [1.2.2 Container](#122-container)
+  * [2 Functionality](#2-functionality)
+      * [2.1 Target Deployment Use Cases](#21-target-deployment-use-cases)
+  * [3 Design](#3-design)
+    * [3.1 Overview](#31-overview)
+    * [3.2 DB Changes](#32-db-changes)
+      * [3.2.1 CONFIG DB](#321-config-db)
+      * [3.2.2 APP DB](#322-app-db)
+      * [3.2.3 STATE DB](#323-state-db)
+      * [3.2.4 ASIC DB](#324-asic-db)
+      * [3.2.5 COUNTER DB](#325-counter-db)
+    * [3.3 User Interface](#33-user-interface)
+      * [3.3.1 REST API Support](#332-rest-api-support)
+      * [3.3.2 gNMI Support](#333-gnmi-support)
+  * [4 Flow Diagrams](#4-flow-diagrams)
+  * [5 Error Handling](#5-error-handling)
+  * [6 Unit Test Cases](#6-unit-test-cases)
+    * [6.1 Functional Test Cases](#61-functional-test-cases)
+    * [6.2 Negative Test Cases](#62-negative-test-cases)
+  
+# List of Tables
+[Table 1: Abbreviations](#table-1-abbreviations)
+[Table 2: OC YANG SONiC YANG Mapping](#4-flow-diagrams)
+
+# Revision
+| Rev |     Date    |       Author          | Change Description                |
+|:---:|:-----------:|:---------------------:|-----------------------------------|
+| 0.1 | 02/24/2024  | Satoru Shinohara | Initial version                   |
+
+# About this Manual
+This document provides general information about the OpenConfig configuration of portchannel (aggregate) interface in SONiC.
+
+# Scope
+- This document describes the high level design of OpenConfig configuration of portchannel interfaces via REST and gNMI in SONiC.
+- This does not cover the SONiC KLISH CLI.
+- This covers only the portchannel interfaces configuration.
```

Comment:
Don't we support counters?

Reply:
We do support counters for interfaces + Ethernet.
However, when comparing E-SONiC vs SONiC-net's DB's on a fresh image, I saw that creating a PortChannel on Community-SONiC Switch did not produce PortChannel in the COUNTERS_DB in a way that was expected. 

E-SONiC Showed portchannel inside **COUNTERS_PORT_NAME_MAP** 

>   "COUNTERS_PORT_NAME_MAP": {
>     "expireat": 1719361893.0460842,
>     "ttl": -0.001,
>     "type": "hash",
>     "value": {
>     ...
>           "PortChannel105": "oid:0x2000000000bce"
>     }
>   },

While Community-SONiC Showed it inside **COUNTERS_LAG_NAME_MAP**

>   "COUNTERS_LAG_NAME_MAP": {
>     "expireat": 1719362054.8109653,
>     "ttl": -0.001,
>     "type": "hash",
>     "value": {
>       "": "",
>       "PortChannel10": "oid:0x2000000000c41"
>     }
>   },

Following the OID for both, E-SONiC Produced the proper counters table

>   "COUNTERS:oid:0x2000000000bce": {
>     "expireat": 1719361893.0559826,
>     "ttl": -0.001,
>     "type": "hash",
>     "value": {
>       "PORT_STAT_IF_IN_BITS_PER_SECOND": "0",
>       "PORT_STAT_IF_IN_OCTETS_PER_SECOND": "0",
>       "PORT_STAT_IF_IN_PKTS_PER_SECOND": "0",
>       ...

While Community-SONiC Showed No Results 
![image](https://github.com/nagarwal03/SONiC-oc-yang-intf/assets/146251176/ba952d05-3f67-4f20-aec5-f30f08b80882)

Reply:
Once a traffic flows on the given PO interface, the counters should be populated for the OID. 
e.g. mock tables to test LAG counters 
https://github.com/sonic-net/sonic-utilities/blob/0e6a55ef5eac306ef61d6f0241625a6baee42ab8/tests/mock_tables/asic1/counters_db.json

Reply:
But, I don't think you code can handle counters because your callback does not have a translation logic for those attributes from Counter DB to YANG. You can check.

Reply:
I just checked that those attributes for LAG counters - SAI_ROUTER_INTERFACE_STAT_IN_ERROR_OCTETS etc. -  are not presented in OC yang. So, it's ok.